### PR TITLE
Log out functionality and auth improvements

### LINKED ISSUE #37: Integrate GitHub login/logout with our auth screen
## Description
User should be able to:
- log in by clicking the button
- receive error messages if something went wrong, the app must not crash
- log out after they have logged in

## Correct issue checklist:
- [x] I have added a detailed description
- [x] I have specified the environment
- [x] I have assigned a Team label (front-end / node / git / design / gatsby)
- [x] I have assigned a Description label (enhancement / bug / etc.)
- [x] I have assigned a Project
- [x] I have added a Milestone (if applicable)

### PR DESCRIPTION
**Pull request by karczewskiPiotr**
## Description
This PR implements log out functionality, improves session handling and refactors auth.

In case of no environmental variables present GitHub will basically show this instead of log in form. 

image

There is no real good way of validating env variables as validating their presence does nothing for us, We'd need to check against GitHub if they are valid but there is no such functionality. So just make sure you have variables as the end user is bound to have them after deployment anyway.
## Linked issues
closes #37 

## Correct PR Checklist
- [x] I have linked an issue (use e.g. `closes #1`)
- [x] I have assigned Reviewers
- [x] I have assigned a Team label
- [x] I have assigned an Description label
- [x] I have assigned a Project

### CODE REVIEW

**Review by 217853 — APPROVED** [review]

File: src/main/index.ts [hunk 1]
```diff
@@ -3,6 +3,8 @@ import { configStore } from '../shared/configureStore';
 declare const MAIN_WINDOW_WEBPACK_ENTRY: any;
 declare const MAIN_WINDOW_PRELOAD_WEBPACK_ENTRY: any;
 
+require('dotenv').config();
```
Comment (lukasz-starosta) [inline src/main/index.ts:6]: Is this still needed?

File: src/renderer/components/NavigationBar/NavigationBar.tsx [hunk 2]
```diff
@@ -67,7 +68,7 @@ const NavigationBar = () => {
         {renderListOfButtons(BOTTOM_BUTTONS)}
         <button
           className='navbar__button'
-          onClick={() => console.log('TODO: implement logout')}
+          onClick={() => dispatch(terminateSessionAsync())}
```
Comment (lukasz-starosta) [inline src/renderer/components/NavigationBar/NavigationBar.tsx:71]: Will this redirect us to the login screen too?

File: src/renderer/components/RouterComponents/Description/Description.tsx [hunk 3]
```diff
@@ -26,19 +15,12 @@ const Description = () => {
...
-      <button
-        onClick={() =>
-          dispatch(displayUserData(currentUser.auth.accessToken.value))
-        }
-      >
-        Display user data
-      </button>
       <p>
-        Welcome <b>{currentUser.data.nick}</b>!
+        Welcome <b>{currentUser.data?.nick || ''}</b>!
```
Comment (lukasz-starosta) [inline src/renderer/components/RouterComponents/Description/Description.tsx:19]: The correct syntax for optional chaining would be `currentUser.data?.nick ?? ''`

**Review by lukasz-starosta — COMMENTED** [review]
Please respond to the issues in comments, I think it's alright just want to make it clear

File: src/main/index.ts (hunk 1 above)
Comment (karczewskiPiotr) [reply src/main/index.ts:6]: yep, renderer and main don't share them. previously env variables were passed down from renderer in functions calling API which just makes them have more params than necessary

File: src/renderer/components/NavigationBar/NavigationBar.tsx (hunk 2 above)
Comment (karczewskiPiotr) [reply src/renderer/components/NavigationBar/NavigationBar.tsx:71]: yes 👌

File: src/renderer/components/RouterComponents/Description/Description.tsx (hunk 3 above)
Comment (karczewskiPiotr) [reply src/renderer/components/RouterComponents/Description/Description.tsx:19]: Cool stuff, didn't know about nullish coalescing operator before. 
Either way this was just placeholder when working and isn't really necessary as data is already there when this screen is rendered and in the end will be removed so won't bother.

**Review by lukasz-starosta — APPROVED** [review]

**Pull request merged**